File: ankor-core/src/main/java/at/irian/ankor/change/ChangeEvent.java
package at.irian.ankor.change;

import at.irian.ankor.event.EventListener;
import at.irian.ankor.event.ModelPropertyEvent;
import at.irian.ankor.event.source.Source;
import at.irian.ankor.ref.Ref;

/**
 * @author Manfred Geiler
 */
public class ChangeEvent extends ModelPropertyEvent {
    //private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(ChangeEvent.class);

    private final Change change;

    public ChangeEvent(Source source, Ref changedProperty, Change change) {
        super(source, changedProperty);
        this.change = change;
    }

    public Ref getChangedProperty() {
        return getProperty();
    }

    public Change getChange() {
        return change;
    }

    @Override
    public boolean isAppropriateListener(EventListener listener) {
        return listener instanceof ChangeEventListener;
    }

    @Override
    public void processBy(EventListener listener) {
        ((ChangeEventListener)listener).process(this);
    }

    @Override
    public String toString() {
        return "ChangeEvent{" +
               "changedProperty=" + getChangedProperty() +
               ", change=" + change +
               '}';
    }
}

File: #ANKOR.net/Ankor.Core/Change/ChangeEvent.cs
using Ankor.Core.Event;
using Ankor.Core.Ref;

namespace Ankor.Core.Change {
	public class ChangeEvent : ModelEvent, IModelEvent {
		public IRef Property { get; private set; }
		public Change Change { get; private set; }

		public ChangeEvent(IEventSource source, IRef property, Change change) : base(source) {
			Change = change;
			Property = property;
		}

		public override bool IsAppropriateListener(IModelEventListener<IModelEvent> listener) {
			return listener is ModelEventListener<ChangeEvent>;
		}

		public override void ProcessBy(IModelEventListener<IModelEvent> listener) {
			((ModelEventListener<ChangeEvent>)listener).Process(this);
		}
	}
}